mpdev/internal/tf/overwrite.go

```diff
@@ -142,3 +147,64 @@ func overwriteFile(filename string, varname string, value string) error {
 	_, err = f.Write(file.BuildTokens(nil).Bytes())
 	return err
 }
+
+// OverwriteMetadata replaces default values for variables in Blueprints Metadata
+//
+// This overwrite rearranges the order of keys in the YAML, since we are not using the
+// definition in
+// https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/cli/bpmetadata/types.go
+//
+// We are not using this definition because
+// 1. There are version compatiblilty issues with Kpt and cloud-foundation-toolkit to resolve
+// 2. We will avoid dropping fields if mpdev is using an out-of-date version of cloud-foundation-toolkit
+func OverwriteMetadata(config *overwriteConfig, dir string) error {
+	fmt.Printf("Replacing the default values of the variables: %s in %s\n",
+		config.Variables, metadataFile)
+
+	data, err := os.ReadFile(path.Join(dir, metadataFile))
+	if err != nil {
+		// CLI only modules will not have a metadata file. Ignore file not found errors
+		if os.IsNotExist(err) {
+			return nil
+		}
```

Comment:
it seems like this will continue and try to read data if it's not a file not found error. do we need to return an error?